sapl-lang/src/main/java/io/sapl/interpreter/DefaultSAPLInterpreter.java

```diff
@@ -69,14 +69,28 @@ public SAPL parse(String saplDefinition) {
 
     @Override
     public SAPL parse(InputStream saplInputStream) {
-        var sapl       = loadAsResource(saplInputStream);
+
+        InputStream convertedSaplInputStream;
+        try {
+            convertedSaplInputStream = InputStreamHelper.detectAndConvertEncodingOfStream(saplInputStream);
+        } catch (IOException e) {
+            log.error(e.getMessage(), e);
+            throw new PolicyEvaluationException(
+                    composeReason("Invalid byte sequence in InputStream. Could not transform to UTF-8"));
```

Comment:
Add cause exception to history

Reply:
I hope that my updated PR resolves this issue. I'm unable to test it beforehand due to SonarLint issues.